src/app.js

```diff
@@ -6,8 +6,318 @@ import _ from 'underscore';
+//MODELS & COLLECTIONS IMPORT
+import TripsList from './app/collections/trips_list.js';
+import Trip from './app/models/trip.js';
+import Reservation from './app/models/reservation.js';
 
-$(document).ready( () => {
-  $('main').html('<h1>Hello World!</h1>');
+////////////VARIABLE DECLARATIONS AND FUNCTIONS ///////////////
+const TRIP_FIELDS = ['name', 'continent', 'about', 'category', 'weeks', 'cost'];
+
+const RES_FIELDS = ['name', 'age', 'email'];
+
+let tripsTemplate;
+let tripTemplate;
+let statusMessageTemplate;
+
+const tripsList = new TripsList();
+
+////////////////////REPORT ERROR STATUS ////////////////////
+const reportStatus = function reportStatus(status, message) {
+  console.log(`Reporting ${ status } message: ${ message}`);
+  const generatedHTML = statusMessageTemplate({
+    status: status,
+    message: message,
+  });
+
+  $('#status-messages ul').append(generatedHTML);
+  $('#status-messages').show();
+};
+
+const filterTrips = function filterTrips(trips) {
+  let filter = $('#filter-value').val().toLowerCase().trim();
+  let filterType = $('#filter-type').val();
+
+  // alert(filterType); // way to test that the input is what we think it will be
+  if (filter === "") {
+    return trips;
+  }
+
+  // want to pass in collection.models -- want to pass in an array of models
+  let results = _.filter(tripsList.models, function(trip) {
+    if (filterType === 'name' || filterType === 'category' || filterType === 'continent') {
+      if (trip.get(filterType).toLowerCase().includes(filter)) {      // string filter
+        return true;
+      }
+    } else {
+      if (trip.get(filterType) <= parseInt(filter)) {   // integer filter
+        return true;
+      }
+    }
+
+    return false;
+  });
+
+  return results;
+}
+
+/////////////////RENDER ALL TRIPS //////////////////
+const renderTrips = function renderTrips(list) {
+  let trips = filterTrips(list);
+
+  $('#trips-table').show();
+  const tripTableElement = $('#trips-list');
+  tripTableElement.html('');
+
+  trips.forEach((trip) => {
+    const generatedHTML = tripsTemplate(trip.attributes);
+    tripTableElement.append(generatedHTML);
+  });
+
+  $('.see-trips-button').hide();
+};
+
+//////////// RENDER INDIVIDUAL TRIPS /////////////////
+const renderTrip = function renderTrip(trip) {
+  const tripElement = $('#trip-detail');
+
+  const generatedHTML = $(tripTemplate(trip.attributes));
+  generatedHTML.on('click', '.button', addReservationHandler);
+  tripElement.html(generatedHTML);
+};
+
+////////////// ADD TRIP HANDLER //////////////////////
+const addTripHandler = function(event) {
+  event.preventDefault();
+
+  const tripData = {};
+  TRIP_FIELDS.forEach((field) => {
+    // select the input corresponding to the field we want
+
+    if (field === "continent") {
+      const selectElement = $(`#add-trip-form option[name="${ field }"]`);
+      let selectValue = selectElement.val();
+
+      tripData[field] = selectValue;
+      selectElement.val('');
```

Comment:
This clears the value of the selected item.  The text remains but the value goes away.   I would suggest instead clearing the values only after a successful save.